### PR TITLE
River/seller with 0 eth onboarding 2

### PR DESCRIPTION
**Pull request by riverfor**
I have updated the [PR](https://github.com/OriginProtocol/origin/pull/784) to fix the "seller with 0 eth onboarding (#661)"  issue.

If this PR is merged please delete the old [PR](https://github.com/OriginProtocol/origin/pull/784) also.

Thanks.

### CODE REVIEW

**Comment by jordajm** [comment]
@micahalcorn I like the idea of this PR. 

A couple of thoughts:

1) When I try to navigate away from the listing-create flow, I get the warning that says "if you leave this page, your progress will be lost". This should probably be removed since it's no longer accurate.

2) We'll need to test this with fractional listings. If it's merged before my [current fractional PR](https://github.com/OriginProtocol/origin/pull/1107), I'll test it out in my branch and fix any funkiness.

3) There is a small issue with editing listings. When click "edit", it shows the "You have an unfinished listing" modal. Aside from that, it seems to work.

File: origin-dapp/src/components/listing-create.js [hunk 1]
```diff
@@ -1385,6 +1412,42 @@ class ListingCreate extends Component {
                 </div>
               </Modal>
             )}
+            <Modal backdrop="static" isOpen={showEthNotEnough}>
+              <div className="image-container">
+                <img src="images/flat_cross_icon.svg" role="presentation" />
+              </div>
+              <FormattedMessage
+                id={'eth-not-enough.error1'}
+                defaultMessage={'Your ETH is not enough.'}
```
Comment (sparrowDom) [inline origin-dapp/src/components/listing-create.js]: I would change the error message to something like this:

```
You don’t have enough funds to complete this transaction.
Please add funds to your wallet.
```

**Comment by sparrowDom** [comment]
I like these changes a lot, great work @riverfor. From what I've tested it works nicely, besides @jordajm observations I will just add a couple: 
 screenshot 2018-12-29 02 20 23

Mouse over on above image renders it white, hiding the "Ok" text. 

screenshot 2018-12-29 02 21 38

Continue button should not have a white rim, and hover should darken the blue colour. The wording of the dialog can probably be improved a bit. 

screenshot 2018-12-29 02 40 29

When you click to edit a listing the `CreateListing` `updateState` action is triggered that populates the state in the local storage. It saves an incorrect `selectedSchemaId` that results in the page crash when trying to open a `Create new Listing` form. I think when editing a listing the newly created actions should not be called. 

After the issues are resolved I am all for merging this. Shouldn't be too much work to merge the changes in my ongoing PRs.

**Comment by micahalcorn** [comment]
Thanks guys. 🙏 @riverfor can you make these changes soon?

**Comment by riverfor** [comment]
@jordajm @sparrowDom Thanks for your detail suggestions.

@micahalcorn No problem, I will update it today. 

BTW: Happy new year.

**Comment by riverfor** [comment]
@micahalcorn I have updated it already.

**Comment by micahalcorn** [comment]
@riverfor thanks again for cleaning this up. 😑 I will try to get to this today so that it's the last time.

**Comment by micahalcorn** [comment]
@riverfor I've made some changes to the copy and styles for the draft-related UI. I've also removed the ETH-detection since it would be fairly complicated to calculate the amount of required gas using the existing code structure. We've also decided to let funding be a wallet concern for now. Both MetaMask and Origin Wallet will display and appropriate message if there is not enough ETH. The main thing is just to ensure that the seller won't lose her work, as you have done here.

I did notice that there is some issue when editing a listing related to the JSON schema path. `selectedSchemaId` is coming in as an absolute path instead of just the id, so then it throws an exception. It also appears that there are several places where `this.state` is being used for boost-(OGN)-related logic, and breaking step 3. Do you mind taking another look at this so that we don't break some existing features?

**Comment by jordajm** [comment]
@riverfor regarding the`selectedSchemaId` being an absolute path as @micahalcorn mentioned, I think it's probably fine that it's an absolute path, but you might need to use this `schemaAdapter` util to fetch the schema for the listing type and render the "details" form:
https://github.com/OriginProtocol/origin/blob/master/origin-dapp/src/utils/schemaAdapter.js

Since you're pulling the listing data from local storage instead of calling [our `getListing` method](https://github.com/OriginProtocol/origin/blob/master/origin-dapp/src/utils/listing.js#L308-L323) like we currently do for editing a listing, I think you'll have to call `schemaAdapter` directly so it swaps out the hostname of the schema URL correctly based on environment, handles schema versioning, etc.

Feel free to ping me if you have any questions about this part of the work. Thanks again for your contributions on this - it's looking great.

**Comment by riverfor** [comment]
OK, I will take a look as soon.

**Comment by riverfor** [comment]
@micahalcorn @jordajm 

Hi, guys, I have made some changes, would you take a look ?

Thanks

**Comment by riverfor** [comment]
@micahalcorn @jordajm 

Can you guys take a look?
Thanks

**Comment by micahalcorn** [comment]
@riverfor there is an issue with updating a listing:

1. Click to edit a listing
2. Make changes
3. Navigate away from listing form
4. Confirm that you want to leave
5. Click "Add a listing" and resume editing
6. Click done and get `Listing id undefined must be a string`

**Comment by riverfor** [comment]
@micahalcorn @jordajm 

Can you guys take a look? Sorry for late update, it's Chinese New Year half a month before.

Thanks

**Comment by micahalcorn** [comment]
Thanks @riverfor 🇨🇳

**Pull request merged**